### DISCUSSION TITLE
Add the element in schema dynamically based on selection?

### DISCUSSION THREAD

**Discussion by yangbin990** [vueform/vueform#363] [Questions]
Hi Bro,

I created a simple form that is always generated based on a schema for debugging and testing purposes. The schema is defined in a textarea component.

I want to dynamically add a new element to this schema in the textarea when "First Class" is selected. My challenge is figuring out how to reference the textarea element in the onChange event.
  
![VueFormQuestion1](https://github.com/user-attachments/assets/a63bf243-a2df-46e7-94b4-7c6537536a73)

I got a dummy code like below
```
class: {
	type: 'select', 
	items: [ 
		'Economy',
		'Premium economy',
		'Business',
		'First class',
	], 
	onChange(newValue, oldValue, el$) {
		if (newValue === 'First class')
		{ 
			**jsonString.push('....');**
		} 
	}
}
```

Thanks for your time.

Bin

**Comment by adamberecz** [comment]
Have you considered using [conditions](https://vueform.com/docs/conditional-rendering#conditional-rendering)?

**Reply by yangbin990** [reply]
Condition would be working with show/hide the sub elements. My difficulty is how to reflect the current vueform schema back to my textarea, that textarea is not part of vueform. 

Bascially when the class selection value is changed, it will grab the sub elements using an API and merge the content with current vueform schema, vueform is then updated, the textarea would show the new vueform json schema. when anything goes wrong, I can log that schema textarea for further investigation.

Thanks

**Reply by adamberecz** [reply]
You can dynamically change the `schema` by eg. defining it as a `computed`. Then in the computed you can rely on other things that determine the final schema.
